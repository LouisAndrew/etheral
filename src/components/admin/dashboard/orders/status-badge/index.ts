import { StatusBadge, BadgeTypes } from './status-badge';

export { BadgeTypes };

export default StatusBadge;
