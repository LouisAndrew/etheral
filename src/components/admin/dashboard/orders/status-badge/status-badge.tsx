import React from 'react';

import { Text } from 'rebass';

import { getDate } from 'helper/get-date';

type Props = {
    type: BadgeTypes;
    paid?: boolean;
    shipped?: boolean;
    date?: Date;
};

export enum BadgeTypes {
    PAYMENT,
    SHIPPING,
    DATE,
}

const StatusBadge: React.FC<Props> = ({
    type,
    paid,
    shipped,
    date,
    ...rest
}) => {
    if (type === BadgeTypes.PAYMENT && paid !== undefined) {
        return (
            <Text
                m={1}
                sx={{ borderRadius: 3 }}
                variant={paid ? 'statusBadgePaid' : 'statusBadgeNotPaid'}
                {...rest}
            >
                {paid ? 'Paid' : 'Not Paid'}
            </Text>
        );
    } else if (type === BadgeTypes.SHIPPING && shipped !== undefined) {
        return (
            <Text
                m={1}
                sx={{ borderRadius: 3 }}
                variant={
                    shipped ? 'statusBadgeShipped' : 'statusBadgeNotShipped'
                }
                {...rest}
            >
                {shipped ? 'Shipped' : 'Not Shipped'}
            </Text>
        );
    } else if (type === BadgeTypes.DATE && date !== undefined) {
        return (
            <Text
                m={1}
                sx={{ borderRadius: 3 }}
                variant={'statusBadgeDate'}
                {...rest}
            >
                Order date: {getDate(date)}
            </Text>
        );
    } else {
        return <h2>WTF</h2>;
    }
};

export { StatusBadge };
