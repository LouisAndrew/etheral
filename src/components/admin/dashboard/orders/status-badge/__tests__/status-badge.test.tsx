import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

// import { render, cleanup } from '@testing-library/react'
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { StatusBadge, BadgeTypes } from '../status-badge';

describe('StatusBadge', () => {
    const Element = <StatusBadge type={BadgeTypes.PAYMENT} paid={false} />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    /* it('renders correctly', () => {
		const { getByTestId } = render()
	}) */

    it('matches snapshot', () => {
        const run = true;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
