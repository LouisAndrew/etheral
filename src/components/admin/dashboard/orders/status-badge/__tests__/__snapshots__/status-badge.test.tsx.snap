// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBadge matches snapshot 1`] = `
<div
  className="css-1a0lovt"
>
  Not Paid
</div>
`;
